### PR TITLE
Adding an API to get the shape of the Singleton Type

### LINKED ISSUE #36995: [Semantic API] Need an API to get the shape of the SingletonType
**Description:**
Task of https://github.com/ballerina-platform/ballerina-lang/issues/35783#issuecomment-1181626283

**Describe your problem(s)**
There is no way to get the shape of the internal type of singleton-type symbols.
Ex: get the shape as Integer, Float, String

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/35783

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #36995

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSingletonTypeSymbol.java [hunk 1]
```diff
@@ -53,6 +55,10 @@ public BallerinaSingletonTypeSymbol(CompilerContext context, BLangLiteral shape,
         } else {
             this.typeName = shape.toString();
         }
+
+        // Getting the original type of the shape
+        TypesFactory typesFactory = TypesFactory.getInstance(this.context);
+        originalType = typesFactory.getTypeDescriptor(shape.getBType());
```
Comment (dulajdilshan) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSingletonTypeSymbol.java:61]: It is not good to create a type-factory instance inside the constructor. Why don't we pass the original type from the outside?

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaConstantSymbol.java [hunk 2]
```diff
@@ -162,6 +162,11 @@ private String toStringVal(Object obj, BType valType) {
         return String.valueOf(obj);
     }
 
+    @Override
+    public TypeSymbol originalType() {
```
Comment (dulajdilshan) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaConstantSymbol.java:166]: We don't have to add `originalType()` API in constant symbols right?

Comment (sanjana) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaConstantSymbol.java:166]: @dulajdilshan, `BallerinaConstantSymbol` class implements the `ConstantSymbol` interface which extends the `SingletonTypeSymbol` interface. Therefore, when a new API method is exposed via the `SingletonTypeSymbol` interface, its implementation should also be made available in the `BallerinaConstantSymbol` class. 

As per a discussion held with @mohanvive, it was decided to proceed with providing the `broaderTypeDescriptor` for constants which aligns with singleton's original type.

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaConstantSymbol.java:166]: Ah okay. I noticed `ConstantSymbol` extends `SingletonType`.

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Pull request merged**